cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/queue/AbstractQueueAdmin.java

```diff
@@ -0,0 +1,111 @@
+   * @return namespace id that this queue belongs to
+   */
+  public static String getNamespaceId(String queueTableName) {
+    // last three parts are namespaceId, appName and flow
+    String[] parts = queueTableName.split("\\.");
+    String namespaceId;
+    if (parts.length == 6) {
+      // cdap.<namespace>.system.queue.<app>.<flow>
+      namespaceId = parts[1];
+    } else {
+      throw new IllegalArgumentException(String.format("Unexpected format for queue table name. " +
+                                                         "Expected 'cdap.<namespace>.system.queue.<app>.<flow>'" +
+                                                         "Received '%s'",
+                                                       queueTableName));
+    }
+    return namespaceId;
+  }
+
+  /**
+   * @param queueTableName actual queue table name
+   * @return app name this queue belongs to
+   */
+  public static String getApplicationName(String queueTableName) {
+    // last three parts are namespaceId (optional - in which case it will be the default namespace), appName and flow
+    String[] parts = queueTableName.split("\\.");
+    return parts[parts.length - 2];
+  }
+
+  /**
+   * @param queueTableName actual queue table name
+   * @return flow name this queue belongs to
+   */
+  public static String getFlowName(String queueTableName) {
+    // last three parts are namespaceId (optional - in which case it will be the default namespace), appName and flow
+    String[] parts = queueTableName.split("\\.");
+    return parts[parts.length - 1];
+  }
+
+
+  /**
+   * This determines the actual table name from the table name prefix and the name of the queue.
+   * @param queueName The name of the queue.
+   * @return the full name of the table that holds this queue.
+   */
+  public String getActualTableName(QueueName queueName) {
+    if (queueName.isQueue()) {
+      // <root namespace>.<queue namespace>.system.queue.<app>.<flow>
+      return getTableNameForFlow(queueName.getFirstComponent(),
+                                 queueName.getSecondComponent(),
+                                 queueName.getThirdComponent());
+    } else {
+      throw new IllegalArgumentException("'" + queueName + "' is not a valid name for a queue.");
+    }
+  }
+
+  protected String getTableNameForFlow(String namespaceId, String app, String flow) {
+    String tablePrefix = getTableNamePrefix(namespaceId);
+    String tableName = tablePrefix + "." + app + "." + flow;
+    return HBaseTableUtil.getHBaseTableName(tableName);
+  }
+
+  public String getTableNamePrefix(String namespaceId) {
+    // returns String with format:  '<root namespace>.<namespaceId>.system.(stream|queue)'
+    String tablePrefix = namespace.namespace(Id.DatasetInstance.from(namespaceId,
+                                                                     unqualifiedTableNamePrefix)).getId();
+    return HBaseTableUtil.getHBaseTableName(tablePrefix);
```

Comment:
Incase someone wants to use getTableNamePrefix method directly (for the purpose of deleting certain tables, for intstance). Since they're going to do a .startsWith(), the tablePrefix also needs to be encoded.